supervisor.py

```diff
@@ -0,0 +1,158 @@
+"""
+Red-DiscordBot built in Supervisor
+Coded by CCubed (AKA Rory in DAPI)
+"""
+import os
+import subprocess
+import sys
+import urllib.request
+from html.parser import HTMLParser
+from zipfile import *
+
+
+IS_X8664 = sys.maxsize > 2**32
+
+
+class LinkParser(HTMLParser):
+    def __init__(self):
+        self.static_download = None
+        super().__init__()
+
+    def handle_starttag(self, tag, attrs):
+        if tag == 'a':
+            for attr in attrs:
+                if attr[0] == 'href':
+                    if 'win32-static.zip' in attr[1]:
+                        self.static_download = attr[1]
+
+
+def check_for_git():
```

Comment:
> mode is a permission mask passed to os.access(), by default determining if the file exists and executable.

with a default of F (existance) and X (executability) permissions

and for os.access

> Use the real uid/gid to test for access to path.

so it *should* check the executability as well for the current user, I could be misunderstanding what I'm reading though